### ISSUE TITLE
Bug: (Unexpected new line between function and parentheses when no function is declared)

### ISSUE THREAD

**Issue by DazKins** [eslint/eslint#16183]
### Environment

Node version: v16.16.0
npm version: v8.11.0
Local ESLint version: Not found
Global ESLint version: v8.21.0 (Currently used)
Operating System: darwin 21.5.0

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```json
{
  "extends": [
    "eslint:recommended"
  ],
  "parserOptions": {
    "ecmaVersion": 11,
    "sourceType": "module"
  },
  "env": {
    "es6": true,
    "mocha": true,
    "node": true,
    "commonjs": true
  },
  "rules": {
    "semi": ["warn", "never"],
    "indent": ["warn", 2, { "SwitchCase": 1 }],
    "space-before-function-paren": ["warn", "always"],
    "quotes": ["error", "double"]
  }
}

```
</details>

I declared a boolean variable and then on the next line direct called an in-line promise.

This is a minimal reconstruction of my scenario:

```js
let b = false
(async () => {})()
```

### What did you expect to happen?

ESLint should accept this as valid with my configuration.

### What actually happened?

ESlint said this was invalid because:

`error  Unexpected newline between function and ( of function call  no-unexpected-multiline`

It appears to be treating the `false` keyword as if it were a function and the open parenthesis on the next line as the start of the parameter list of that function.

Note that the issue was fixed by putting `;` after the boolean declaration.

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by DazKins** [comment]
Just realised when this code executed that nodejs also reject it saying: `false is not a function` which is similar to what I described above.

Perhaps ESLint isn't so wrong about this then, but the error message could be more accurate here.

**Comment by mdjermanovic** [comment]
This works as intended because `false(async () => {})` is a function call. If the current message is causing confusion, maybe we could change it to `Unexpected newline between callee and ( of function call`?

**Comment by nzakas** [comment]
I don't think making a change to the wording would improve the situation. It will still be a bit vague because we don't necessarily know what the stuff coming before the `(` was intended to be. It looks like a function call, that's all we know.

**Comment by nzakas** [comment]
Closing due to age
